ufl/core/base_form_operator.py

```diff
@@ -182,4 +184,27 @@ def __hash__(self):
 
     def __eq__(self, other):
         """Check for equality."""
+        if isinstance(other, Number) and other == 0:
+            return self.empty()
         raise NotImplementedError()
```

Comment:
@dham in [adjoint.py](https://github.com/FEniCS/ufl/blob/02cc03a7ce5f9589e1d92639044b3f33830ad7d7/ufl/adjoint.py#L44) we check whether the BaseForm that we are adjoining is zero. In particular, we need this for `BaseFormOperatorDerivative`. Could we implement equality against 0 generically in `BaseFormOperator`?

Reply:
Quite possibly.